x86.c

```diff
@@ -1,12 +1,27 @@
 // x86 codegen
 
-#include "exe.c"
+#ifdef i386
```

Comment:
Since architecture and OS are disjoint, I think it would make more sense to have macros for architecture (`i386`, `x86_64`, `arm`) and OS (`linux`, `mac`, `windows`).

Reply:
The symbols `i386`, `arm` etc should be renamed to `target_i386`, `target_arm`, etc because it will allow creating cross compilers (for example generating an `arm` program on a `i386` machine).